### ISSUE TITLE
Label updates, bug traige board

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#4497]
In an effort to improve our project efficiency:
- Added PR labels for action needed, reviewer/author and a 'merge when ready label'
- I simplified/combined category labels and removed unused labels
- Added a 'bug' triage project with bug triage board that can be used to track bugs

### PR Labels
- Added author and reviewer action needed labels. Idea here is we can have a label to say "ok, PR is updated, reviewer person please look at this", or "this has been reviewed, author should comment or make updates"

### Label updates

- simplified categories. Notably 'bug' category is just `category: problem`. The idea is we have three distinct groupings now, "problem", "code improvement", or "feature request" (which is basic any kind of improvement). The "UI problem" and "code rules" distinctions are now gone.

![screenshot from 2019-01-03 17-09-10](https://user-images.githubusercontent.com/12397753/50669497-6bc4d800-0f7a-11e9-95b9-acd2a286ec25.png)

### Bug Triage board

- adding issues to project 'bug triage' will add them to the board.
- https://github.com/triplea-game/triplea/projects/8
![screenshot from 2019-01-03 17-10-12](https://user-images.githubusercontent.com/12397753/50669505-741d1300-0f7a-11e9-8774-acbd775226eb.png)

**Comment by DanVanAtta** [comment]
Also open for discussion:
- we could use assignee to solely indicate who has the next action instead of label
- part of the goal is so we can scan the PR board and know which PRs can use attention/action and which are up to date and waiting.

**Comment by DanVanAtta** [comment]
Hah, I've convinced myself that PR assignee is better than reviewer/author labels. I'll delete those two labels for now.

**Comment by ssoloff** [comment]
:+1:

My only suggestion would be to use consistent casing for the PR labels.  Currently, some are Mixed Case and some are UPPERCASE.  I say just go with Mixed Case to not offend anyone's sensibilities. :smile:

**Comment by DanVanAtta** [comment]
@ron-murhammer  @RoiEXLab @ssoloff 

RE: Bug Triage board

I think it makes sense for us to use the board to prioritize bugs. We can stack rank them relative to another, after triage we put bugs into either the 'urgent' backlog, or the normal backlog. Within each backlog, we can keep each list sorted so that the bugs higher in the list are more important than the ones lower in the list. It's most important we keep this list sorted to around the top 10 elements, and the ones below can be a bit more arbitrarily ordered.

If this kind of process sounds useful, I think it could be to add focus on bugs and try to get the right fixes in sooner, I'd be happy to volunteer to write this process up for our documentation.

**Comment by ron-murhammer** [comment]
@DanVanAtta Makes sense and I did something similar over a year ago with one of the github extensions (can't remember the name at the moment). The 2 biggest problems with it last time were 1. not everyone using it (so mostly just me maintaining and looking at it) and 2. once you get to a certain number of issues it gets somewhat unwieldy (if I have a backlog of 50+ issues its hard to prioritize which are even the 10 most important especially cause it ends up being somewhat up to people's opinions). It probably helps some now that most feature requests are tracked in the forum to reduce the sheer number of issues.

I think the main question is do we think we'll get enough value out of it and will it be maintainable enough to put time into using it? I'd say today we primarily use a label for high priority production issues, use assignee when one of us is working on an issue, and tend to look at the most recently created/update issues. That is a very basic system but also very simple and easy to use.

**Comment by RoiEXLab** [comment]
I'll try to use it 👍

**Comment by DanVanAtta** [comment]
Good points raised.  I'm hoping we can avoid a number of the problems with the previous board, which was before Github had yet to worked out projects and labels.

Beyond the project board being integrated, I think process will help.

One thing to note, issues are included on the board by ***project***, _not_ by _label_. By convention, anything that does not need an immediate fix with the label "category: problem" would be added to the bug triage project. In essence that would be the first step in our process, 'issue triage'.

The next step is then looking at things from the bug board. We can filter for any 'category: problem' labels and add them to the board. For each issue in bug-triage, we likely will just add it to the backlog.  The swim-lanes would look like this:
```
triage  |  backlog |  top-10  | in-progress | done
```

To explain the 'top-10', that is the top-10 priority tickets and are always kept in priority ranked order. Any issue from triage that is of more importance can displace an existing top-10, otherwise it is added to the backlog. Whenever we complete a top-10, we'll move a ticket from the top of the backlog to the bottom of top-10.

I'm hoping this will work well enough we could resurrect the ice-box tickets. In the meantime, thoughts? Any glaring holes in this process?

**Comment by RoiEXLab** [comment]
@DanVanAtta Being used to a SCRUM workflow at work, your proposal seems reasonable 👍

**Comment by DanVanAtta** [comment]
Another suggestion, let's use 'assignee' to mean 'in-progress'. If something is assigned, we can then assume that a person is making progress against an issue. We probably should try to favor self-assigning so that we are properly volunteering when we have the capacity available. 

I think that means we can get rid of the 'in-progress' column from the swim lanes. I also think we can get rid of the 'triage' column and instead add things more blindly to the backlog. It's easy to sort by issue number, which matches to chronological, and we'll move things into top-10 if they are more important, otherwise it's a FIFO queue. Thoughts?
